File: 328.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* oddEvenList(ListNode* head) {
        if (head == nullptr) return head;
        
        ListNode* even = head->next; 
        ListNode* f_even = even; 
        ListNode * odd = head; 
        while(even != nullptr){
            if (even->next == nullptr) return head; 
            odd->next = even->next; 
            odd = odd->next; 
            even->next = odd->next; 
            odd->next = f_even; 
            even = even->next; 
      
        }
        
        return head; 
    }
};

File: 328.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head == None or head.next == None:
            return head 
        
        node = head 
        firstE = head.next 
        lastE = head.next 
        firstO = head
        lastO = head 
        
        while lastE != None and lastE.next != None: 
            lastO.next = lastE.next 
            lastO = lastO.next  # advance the lastO 
            prevLastONext = lastO.next 
            lastO.next = firstE 
            lastE.next = prevLastONext 
            lastE = lastE.next
            
            
        
        return firstO
